src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -112,26 +112,29 @@ public TestResultSet GetVsTestResults(NUnitTestEventTestCase resultNode, ICollec
 
             if (testCaseResult != null)
             {
-                if (testCaseResult.Outcome == TestOutcome.Failed || testCaseResult.Outcome == TestOutcome.NotFound)
+                switch (testCaseResult.Outcome)
                 {
-                    testCaseResult.ErrorMessage = resultNode.Failure?.Message;
-                    testCaseResult.ErrorStackTrace = resultNode.Failure?.Stacktrace;
+                    case TestOutcome.Failed:
+                    case TestOutcome.NotFound:
+                    {
+                        testCaseResult.ErrorMessage = resultNode.Failure?.Message;
+                        testCaseResult.ErrorStackTrace = resultNode.Failure?.Stacktrace ?? resultNode.StackTrace;
 
-                    // find stacktrace in assertion nodes if not defined (seems .netcore2.0 doesn't provide stack-trace for Assert.Fail("abc"))
-                    if (testCaseResult.ErrorStackTrace == null)
+                        
+                        break;
+                    }
+                    case TestOutcome.Skipped:
+                    case TestOutcome.None:
+                        testCaseResult.ErrorMessage = resultNode.ReasonMessage;
+                        testCaseResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, resultNode.ReasonMessage));
+                        break;
+                    default:
                     {
-                        string stackTrace = string.Empty;
-                        foreach (XmlNode assertionStacktraceNode in resultNode.Node.SelectNodes("assertions/assertion/stack-trace"))
-                        {
-                            stackTrace += assertionStacktraceNode.InnerText;
-                        }
-                        testCaseResult.ErrorStackTrace = stackTrace;
+                        if (adapterSettings.ConsoleOut > 0 && !string.IsNullOrEmpty(resultNode.ReasonMessage))
```

Comment:
What do you think of making `adapterSettings.ConsoleOut` an enum in the future?

Reply:
Actually.....I have forgotten why it even is an integer.... it could perhaps just be a boolean.